### PR TITLE
Add documentation for VMware vSphere

### PR DESCRIPTION
**Pull request by madchr1st**
The first how-to for OpenShift 4 on VMware vSphere is the pre-install checklist, to ensure you have setup everything correctly before starting the installation.

The main installation how-to will follow.

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+* `api.ocp4-vs.appuio.ch`
+* `console.ocp4-vs.appuio.ch`
+* `*.apps.ocp4-vs.appuio.ch` (wildcard certificate)
+
+[WARNING]
+====
+For the certificates to be valid in modern web browser, make sure the _Subject Alternative Name (SAN)_ contains the hostname of the certificate.
+
+[[additional-ca]]
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:117]: Might be worth to place the part about CA in its own admonition.

File: docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+If the certificates are issued from your own private PKI, the RootCA certificate of your PKI needs to be added to the `install-config.yaml` as described in the https://docs.openshift.com/container-platform/4.4/networking/configuring-a-custom-pki.html#installation-configure-proxy_configuring-a-custom-pki[documentation]:
+[source,bash]
+----
+...
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:121]:
```suggestion
…
```

File: docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+additionalTrustBundle: | 
+    -----BEGIN CERTIFICATE-----
+    <MY_TRUSTED_CA_CERT_1>
+    -----END CERTIFICATE-----
+    -----BEGIN CERTIFICATE-----
+    <MY_TRUSTED_CA_CERT_2>
+    -----END CERTIFICATE-----
+...
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:129]:
```suggestion
…
```

There is a character for this. Just placing three dots, typographically looks odd.

File: docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+|===
+
+Possible further surrounding systems:
+
+* Access to Git repositories from all worker nodes
+* Access to databases and other applications from all worker nodes
+
+=== Outgoing proxy
+
+The following URLs need to be whitelisted on the outgoing proxy, as described in the https://docs.openshift.com/container-platform/4.4/installing/install_config/configuring-firewall.html[documentation].
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:171]:
```suggestion
The following URLs need to be whitelisted on the outgoing proxy, as described at https://docs.openshift.com/container-platform/4.4/installing/install_config/configuring-firewall.html[Configuring your firewall].
```

File: docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+If the connection to the proxy is done via HTTPS, the CA certificate, which was used to sign the certificate of the proxy server, needs to be added to the `install-config.yaml`, as described <<additional-ca,above>>.
+
+See https://docs.openshift.com/container-platform/4.4/installing/installing_vsphere/installing-vsphere.html#installation-configure-proxy_installing-vsphere[Configuring the cluster-wide proxy during installation] for details.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:175]: Placing the part about the CA in its own admonition, allows you to give it a title. You can then reference that title in the link text.

Comment (corvus-ch) [reply docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:175]: https://asciidoctor.org/docs/user-manual/ holds an example of what I mean. Search for _Feeding the Werewolves_.

File: docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+...
+----
+Multiple CA certificates can be added.
+====
+
+=== Load balancers
+
+For accessing API and applications, the following load balancers are required:
+
+* Console load balancer
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:138]: Maybe `API load balancer` would be a more accurate name?
If I understand correctly the console is served via the apps load balancer in OpenShift 4?

File: docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc (hunk 1 above)
Comment (madchr1st) [reply docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:117]: I had it that way before, looked kinda odd.

File: docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc (hunk 6 above)
Comment (madchr1st) [reply docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:138]: You are absolutely right. This is a leftover from 3.11 doc.

File: docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc (hunk 1 above)
Comment (corvus-ch) [reply docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:117]: How so?

Comment (madchr1st) [reply docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc]: I did not use the title before. Added title, looks good now!

File: docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+====
+`<cluster_name>` must match the folder name in vSphere, in which all OpenShift VMs reside, for vSphere storage integration to work correctly.
+====
+
+With the defined variables, the following DNS records are required:
+
+|===
+|Record |Type |Target 
+
+|`api.ocp4-vs.appuio.ch` |A/CNAME |Console Load Balancer
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:83]:
```suggestion
|`api.ocp4-vs.appuio.ch` |A/CNAME |API Load Balancer
```

File: docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+
+|`api-int.ocp4-vs.appuio.ch` |A/CNAME |Console Load Balancer
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:85]:
```suggestion
|`api-int.ocp4-vs.appuio.ch` |A/CNAME |API Load Balancer
```

File: docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+Certificates for the following hostnames need to be created:
+
+* `api.ocp4-vs.appuio.ch`
+* `console.ocp4-vs.appuio.ch`
+* `*.apps.ocp4-vs.appuio.ch` (wildcard certificate)
+
+[WARNING]
+====
+For the certificates to be valid in modern web browser, make sure the _Subject Alternative Name (SAN)_ contains the hostname of the certificate.
```
[the remark this answers is not in the document]
Comment (madchr1st) [reply docs/modules/ROOT/pages/how-tos/vsphere/pre-install-checklist.adoc:117]: I agree, but prefer OpenSSL :smile: 
```suggestion
For the certificates to be valid in modern web browser, make sure the _Subject Alternative Name (SAN)_ contains the hostname of the certificate.
See https://www.switch.ch/pki/manage/request/csr-openssl/[How to create a CSR with OpenSSL] for an example on how to do so.
```

**Comment by tobru** [comment]
Very cool! Thanks a lot @madchr1st for the work and all the other for commenting, reviewing and discussing.

**Pull request merged**